### PR TITLE
Refactor storage batches to use StorageBatchRequest and BatchResult

### LINKED ISSUE #364: Consider other options for Batch requests.
Suggestion was raised by @Capstan 

We could add a `Batch`/`BatchContext`/`BatchHelper` class that gets storage as an input and have the same `delete`/`get`/`update` methods as `BatchRequest.Builder` but instead of returning `this` for chaining return a `Future` ([this](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/util/concurrent/SettableFuture.html) could help with that) and have a `void apply()` method.

Any further calls on such instance should fail after a successful `apply`

### PR DESCRIPTION
**Pull request by mziccard**
This refactors `Storage` batches to mimic the one in `DNS`. This makes use of the common `BatchResult`. This also fixes #364. 

Now building a batch and getting the result looks like:

``` java
StorageBatch batch = storage.batch();
BlobId firstBlob = BlobId.of("bucket", "blob1"));
BlobId secondBlob = BlobId.of("bucket", "blob2"));
// Users can either register a callback
batch.delete(firstBlob).notify(new BatchResult.Callback<Boolean, StorageException>() {
  @Override
  public void success(Boolean result) {
    // deleted successfully
  }

  @Override
  public void error(StorageException exception) {
    // delete failed
  }
});
batch.update(BlobInfo.builder(secondBlob).contentType("text/plain").build());
StorageBatchResult<Blob> result = batch.get(secondBlob);
batch.submit();
// Or get the result
Blob blob = result.get(); // returns get result or throws StorageException
```

@Capstan I was wondering: is the limit of 100 deletes per batch call still required?

/cc @jgeewax

### CODE REVIEW

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * This class holds a single result of a batch call to Cloud Storage.
+ */
+public class StorageBatchResult<T> extends BatchResult<T, StorageException> {
+
+  StorageBatchResult() {
+  }
+
+  @Override
+  protected void error(StorageException error) {
```
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java]: Codacy Issue found: [Overriding method merely calls super](https://www.codacy.com/app/mziccard/gcloud-java/file/2448215776/issues/source?bid=3123103&fileBranchId=3247265#l31)

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    super.error(error);
+  }
+
+  @Override
+  protected void success(T result) {
```
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java]: Codacy Issue found: [Overriding method merely calls super](https://www.codacy.com/app/mziccard/gcloud-java/file/2448215776/issues/source?bid=3123103&fileBranchId=3247265#l36)

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+package com.google.cloud.storage;
+
+import com.google.cloud.BatchResult;
+
+/**
+ * This class holds a single result of a batch call to Cloud Storage.
+ */
+public class StorageBatchResult<T> extends BatchResult<T, StorageException> {
+
+  StorageBatchResult() {
```
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java]: Codacy Issue found: [Document empty constructor](https://www.codacy.com/app/mziccard/gcloud-java/file/2448215776/issues/source?bid=3123103&fileBranchId=3247265#l27)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/BucketTest.java [hunk 4]
```diff
@@ -249,33 +247,35 @@ public void testGet() throws Exception {
...
-    expect(storage.options()).andReturn(mockOptions).times(3);
-    replay(storage);
+    List<BlobId> blobIds = Lists.transform(blobResults, new Function<Blob, BlobId>() {
+      @Override
+      public BlobId apply(Blob blob) {
+        return blob.blobId();
+      }
+    });
+    EasyMock.expect(storage.get(blobIds)).andReturn(blobResults);
+    EasyMock.replay(storage);
```
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/BucketTest.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'EasyMock.replay' due to existing static import 'org.easymock.EasyMock.replay'](https://www.codacy.com/app/mziccard/gcloud-java/file/2448215795/issues/source?bid=3123103&fileBranchId=3247265#l260)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/BucketTest.java [hunk 5]
```diff
@@ -249,33 +247,35 @@ public void testGet() throws Exception {
...
+  public void testGetAllIterable() throws Exception {
+    initializeExpectedBucket(4);
+    expect(storage.options()).andReturn(mockOptions);
+    List<BlobId> blobIds = Lists.transform(blobResults, new Function<Blob, BlobId>() {
+      @Override
+      public BlobId apply(Blob blob) {
+        return blob.blobId();
+      }
+    });
+    EasyMock.expect(storage.get(blobIds)).andReturn(blobResults);
```
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/BucketTest.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'EasyMock.expect' due to existing static import 'org.easymock.EasyMock.expect'](https://www.codacy.com/app/mziccard/gcloud-java/file/2448215795/issues/source?bid=3123103&fileBranchId=3247265#l275)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java [hunk 6]
```diff
@@ -877,25 +862,36 @@ public void testBatchRequestFail() {
...
+    StorageBatchResult<Boolean> deleteResult1 =
+        batch.delete(BUCKET, blobName, Storage.BlobSourceOption.generationMatch(-1L));
+    StorageBatchResult<Boolean> deleteResult2 = batch.delete(BlobId.of(BUCKET, blobName, -1L));
+    StorageBatchResult<Blob> getResult1 =
+        batch.get(BUCKET, blobName, Storage.BlobGetOption.generationMatch(-1L));
+    StorageBatchResult<Blob> getResult2 = batch.get(BlobId.of(BUCKET, blobName, -1L));
+    batch.submit();
+    try {
+      updateResult.get();
+      fail("Expected StorageException");
```
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2448215805/issues/source?bid=3123103&fileBranchId=3247265#l877)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    assertFalse(result.completed());
+    BatchResult.Callback<Boolean, StorageException> callback =
+        EasyMock.createStrictMock(BatchResult.Callback.class);
+    callback.error(ex);
+    EasyMock.replay(callback);
+    result.notify(callback);
+    result.error(ex);
+    try {
+      result.notify(callback);
+      fail("The batch has been completed.");
```
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2448215801/issues/source?bid=3123103&fileBranchId=3247265#l85)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+  public void setUp() {
+    result = new StorageBatchResult<>();
+  }
+
+  @Test
+  public void testSuccess() {
+    assertFalse(result.completed());
+    try {
+      result.get();
+      fail("This was not completed yet.");
```
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2448215801/issues/source?bid=3123103&fileBranchId=3247265#l46)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchTest.java [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+    EasyMock.reset(batchMock);
+    Capture<RpcBatch.Callback<Void>> callback = Capture.newInstance();
+    batchMock.addDelete(EasyMock.eq(BLOB_INFO.toPb()), EasyMock.capture(callback),
+        EasyMock.eq(ImmutableMap.<StorageRpc.Option, Object>of()));
+    EasyMock.replay(batchMock);
+    StorageBatchResult<Boolean> batchResult = dnsBatch.delete(BLOB_ID.bucket(), BLOB_ID.name());
+    assertNotNull(callback.getValue());
+    try {
+      batchResult.get();
+      fail("No result available yet.");
```
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2448215802/issues/source?bid=3123103&fileBranchId=3247265#l98)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchTest.java [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+    } catch (IllegalStateException ex) {
+      // expected
+    }
+    // testing error here, success is tested with options
+    RpcBatch.Callback<StorageObject> capturedCallback = callback.getValue();
+    capturedCallback.onFailure(GOOGLE_JSON_ERROR);
+    try {
+      batchResult.get();
+      fail("Should throw a StorageExcetion on error.");
```
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2448215802/issues/source?bid=3123103&fileBranchId=3247265#l198)

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageImpl.java [hunk 11]
```diff
@@ -677,24 +663,24 @@ public T apply(BatchResponse.Result<T> result) {
...
   }
 
-  private Map<StorageRpc.Option, ?> optionMap(Long generation, Long metaGeneration,
+  private static Map<StorageRpc.Option, ?> optionMap(Long generation, Long metaGeneration,
       Option... options) {
     return optionMap(generation, metaGeneration, Arrays.asList(options));
   }
 
-  private Map<StorageRpc.Option, ?> optionMap(BucketInfo bucketInfo, Option... options) {
+  private static Map<StorageRpc.Option, ?> optionMap(BucketInfo bucketInfo, Option... options) {
```
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageImpl.java]: Codacy Issue found: [Avoid unused private methods such as 'optionMap(BucketInfo,Option...)'.](https://www.codacy.com/app/mziccard/gcloud-java/file/2448729640/issues/source?bid=3123103&fileBranchId=3247265#l675)

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageImpl.java [hunk 12]
```diff
@@ -677,24 +663,24 @@ public T apply(BatchResponse.Result<T> result) {
     return ImmutableMap.copyOf(temp);
   }
 
-  private Map<StorageRpc.Option, ?> optionMap(Option... options) {
+  private static Map<StorageRpc.Option, ?> optionMap(Option... options) {
     return optionMap(null, null, Arrays.asList(options));
   }
 
-  private Map<StorageRpc.Option, ?> optionMap(Long generation, Long metaGeneration,
+  private static Map<StorageRpc.Option, ?> optionMap(Long generation, Long metaGeneration,
```
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageImpl.java]: Codacy Issue found: [Avoid unused private methods such as 'optionMap(Long,Long,Option...)'.](https://www.codacy.com/app/mziccard/gcloud-java/file/2448729640/issues/source?bid=3123103&fileBranchId=3247265#l670)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java (hunk 7 above)
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2448729662/issues/source?bid=3123103&fileBranchId=3247265#l85)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java (hunk 8 above)
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2448729662/issues/source?bid=3123103&fileBranchId=3247265#l46)

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java (hunk 1 above)
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java]: Codacy Issue found: [Overriding method merely calls super](https://www.codacy.com/app/mziccard/gcloud-java/file/2448729637/issues/source?bid=3123103&fileBranchId=3247265#l31)

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java (hunk 3 above)
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java]: Codacy Issue found: [Document empty constructor](https://www.codacy.com/app/mziccard/gcloud-java/file/2448729637/issues/source?bid=3123103&fileBranchId=3247265#l27)

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java (hunk 2 above)
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java]: Codacy Issue found: [Overriding method merely calls super](https://www.codacy.com/app/mziccard/gcloud-java/file/2448729637/issues/source?bid=3123103&fileBranchId=3247265#l36)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/BucketTest.java [hunk 13]
```diff
@@ -249,33 +247,35 @@ public void testGet() throws Exception {
...
-    expect(storage.options()).andReturn(mockOptions).times(3);
-    replay(storage);
+    List<BlobId> blobIds = Lists.transform(blobResults, new Function<Blob, BlobId>() {
+      @Override
+      public BlobId apply(Blob blob) {
+        return blob.blobId();
+      }
+    });
+    expect(storage.get(blobIds)).andReturn(blobResults);
+    EasyMock.replay(storage);
```
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/BucketTest.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'EasyMock.replay' due to existing static import 'org.easymock.EasyMock.replay'](https://www.codacy.com/app/mziccard/gcloud-java/file/2448729656/issues/source?bid=3123103&fileBranchId=3247265#l260)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java (hunk 6 above)
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2448729666/issues/source?bid=3123103&fileBranchId=3247265#l877)

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java (hunk 1 above)
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java]: Codacy Issue found: [Overriding method merely calls super](https://www.codacy.com/app/mziccard/gcloud-java/file/2448903201/issues/source?bid=3123103&fileBranchId=3247265#l31)

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java (hunk 2 above)
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java]: Codacy Issue found: [Overriding method merely calls super](https://www.codacy.com/app/mziccard/gcloud-java/file/2448903201/issues/source?bid=3123103&fileBranchId=3247265#l36)

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java (hunk 3 above)
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java]: Codacy Issue found: [Document empty constructor](https://www.codacy.com/app/mziccard/gcloud-java/file/2448903201/issues/source?bid=3123103&fileBranchId=3247265#l27)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java (hunk 7 above)
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2448903226/issues/source?bid=3123103&fileBranchId=3247265#l85)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java (hunk 8 above)
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2448903226/issues/source?bid=3123103&fileBranchId=3247265#l46)

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageImpl.java (hunk 12 above)
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageImpl.java]: Codacy Issue found: [Avoid unused private methods such as 'optionMap(Long,Long,Option...)'.](https://www.codacy.com/app/mziccard/gcloud-java/file/2448903204/issues/source?bid=3123103&fileBranchId=3247265#l670)

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageImpl.java (hunk 11 above)
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageImpl.java]: Codacy Issue found: [Avoid unused private methods such as 'optionMap(BucketInfo,Option...)'.](https://www.codacy.com/app/mziccard/gcloud-java/file/2448903204/issues/source?bid=3123103&fileBranchId=3247265#l675)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java (hunk 6 above)
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2448903230/issues/source?bid=3123103&fileBranchId=3247265#l877)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchTest.java (hunk 9 above)
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2448903227/issues/source?bid=3123103&fileBranchId=3247265#l98)

**Comment by Capstan** [comment]
@mziccard The 100 deletes per batch is not a strong requirement; it is a guideline. There is a maximum size of 10MB on the batch request payload, in which you could logically fit in more deletion requests. That said, due to some current service implementation details, the service performs better if the batches are split for better distribution. There are also some current scaling implications if you do a quick QPS ramp-up by issuing many simultaneous batches in parallel, so it's better to either slow-start or limit the number of outstanding batches in flight.

**Comment by mziccard** [comment]
@Capstan Just to make sure I understand: following a [previous comment](https://github.com/GoogleCloudPlatform/gcloud-java/issues/365) we limited the number of deletes per batch to 100, if the user adds more we split them into more batches. Are you suggesting we should limit the overall number of operations on each batch (not only deletes)? If so, could you suggest a reasonable number of operations at which it would be better to create a new batch?

**Comment by Capstan** [comment]
@mziccard Yes, I think limiting the overall number of operations on each batch is, at the moment, useful, and for the same reasons mentioned previously. We have not done specific analysis to find out the break-even point for various types of queries. If pressed, I'd say that, in the absence of further information, I would suggest 100. Batch size is about a limitation in load balancing within the batch. The number of in-flight batches is also related to limitations in load balancing across the service… going from zero QPS to millions of (virtual) QPS in a very short span from a narrow geographic location (e.g., one single client) can result in degrade performance.

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java (hunk 1 above)
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java]: Codacy Issue found: [Overriding method merely calls super](https://www.codacy.com/app/mziccard/gcloud-java/file/2469305475/issues/source?bid=3123103&fileBranchId=3247265#l31)

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java (hunk 2 above)
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java]: Codacy Issue found: [Overriding method merely calls super](https://www.codacy.com/app/mziccard/gcloud-java/file/2469305475/issues/source?bid=3123103&fileBranchId=3247265#l36)

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java (hunk 3 above)
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java]: Codacy Issue found: [Document empty constructor](https://www.codacy.com/app/mziccard/gcloud-java/file/2469305475/issues/source?bid=3123103&fileBranchId=3247265#l27)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java [hunk 14]
```diff
@@ -877,25 +885,36 @@ public void testBatchRequestFail() {
...
+    } catch (StorageException ex) {
+      // expected
+    }
+    try {
+      deleteResult1.get();
+      fail("Expected StorageException");
```
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2469305504/issues/source?bid=3123103&fileBranchId=3247265#l906)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/BucketTest.java [hunk 15]
```diff
@@ -34,12 +33,13 @@
...
 import com.google.cloud.storage.Acl.User;
-import com.google.cloud.storage.BatchResponse.Result;
 import com.google.cloud.storage.BucketInfo.AgeDeleteRule;
 import com.google.cloud.storage.BucketInfo.DeleteRule;
+import com.google.common.base.Function;
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
 
-import org.easymock.Capture;
+import org.easymock.EasyMock;
```
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/BucketTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.easymock.EasyMock'](https://www.codacy.com/app/mziccard/gcloud-java/file/2469305494/issues/source?bid=3123103&fileBranchId=3247265#l42)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java (hunk 7 above)
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2469305500/issues/source?bid=3123103&fileBranchId=3247265#l85)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java (hunk 8 above)
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2469305500/issues/source?bid=3123103&fileBranchId=3247265#l46)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchTest.java (hunk 9 above)
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2469305501/issues/source?bid=3123103&fileBranchId=3247265#l98)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchTest.java (hunk 10 above)
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2469305501/issues/source?bid=3123103&fileBranchId=3247265#l198)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java (hunk 7 above)
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2469779061/issues/source?bid=3123103&fileBranchId=3247265#l85)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java (hunk 8 above)
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchResultTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2469779061/issues/source?bid=3123103&fileBranchId=3247265#l46)

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java (hunk 1 above)
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java]: Codacy Issue found: [Overriding method merely calls super](https://www.codacy.com/app/mziccard/gcloud-java/file/2469779036/issues/source?bid=3123103&fileBranchId=3247265#l31)

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java (hunk 2 above)
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java]: Codacy Issue found: [Overriding method merely calls super](https://www.codacy.com/app/mziccard/gcloud-java/file/2469779036/issues/source?bid=3123103&fileBranchId=3247265#l36)

File: gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java (hunk 3 above)
Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/cloud/storage/StorageBatchResult.java]: Codacy Issue found: [Document empty constructor](https://www.codacy.com/app/mziccard/gcloud-java/file/2469779036/issues/source?bid=3123103&fileBranchId=3247265#l27)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchTest.java (hunk 9 above)
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2469779062/issues/source?bid=3123103&fileBranchId=3247265#l98)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchTest.java (hunk 10 above)
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/StorageBatchTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2469779062/issues/source?bid=3123103&fileBranchId=3247265#l198)

File: gcloud-java-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java (hunk 14 above)
Comment (mziccard) [inline gcloud-java-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java]: Codacy Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2469779065/issues/source?bid=3123103&fileBranchId=3247265#l906)

**Pull request merged**